### DISCUSSION TITLE
Parallel Meep installed without conda does not work

### DISCUSSION THREAD

**Discussion by SilvieLuisa** [NanoComp/meep#2632] [Q&A] [answered]
Hi, I am trying Meep in an environment without conda availability. The admin had to install Meep "from source" for me and he claims that he used the flag --with-mpi. However, when I run my script using mpirun, there are just multiple runs of the same program. The mp.count_processors() function returns 1, output from all the processes is printed and there is NO MESSAGE like this: Using MPI version 3.1, 2 processes. 
Does anyone have an idea where to look (what to check) in order to find out what exactly we are missing here? Any library, wrong setting...? Thanks a lot.

**Comment by mariusCZ** [comment, marked as answer]
So I just checked two builds of meep, one with mpi and one without. The one with automatically prints out the amount of processes used and the one without doesn't. This most likely indicates it was not compiled with mpi support.

Another alternative is to set up miniconda, which (as far as I remember) does not require root and then that will provide a conda environment to install meep. However, as far as I know, in the conda repositories meep is wither with no mpi or only with mpich, so that may not be an option either.

Last option is to build it yourself. It can be done without root, but it is likely you'd have to build a lot of dependencies, which can be a bit of a pain. I've done this myself on an HPC system and have a script, but it is unlikely to work on another system.

Could you provide more information on what kind of system you are running, what kind of scheduler, available MPI versions?

**Reply by SilvieLuisa** [reply]
Hi, the admin has re-installed the parallel Meep, and now it is showing the correct number of processors etc. However, the simulations run with more processors hardly finish - it seems something got stuck somewhere, but I cannot see if it is an issue with OpenMPI or Meep or combination of both.... Does anyone have similar experience? Thanks.

**Reply by mariusCZ** [reply]
You'd need to provide code and what output the program is actually giving. Hard to answer without that information.

**Reply by smartalecH** [reply]
If they also installed with openMP, you need to manually set the number of threads to 1. Some flavors of openMP default to using all the threads, which won't work well if you are also using MPI.

**Comment by SilvieLuisa** [comment]
The issue has been resolved with installing Miniconda. One of the issues might have been that the MPI version had been old. But I really do not recommend manual installation ;-)
